### PR TITLE
🚧Add circle rendering

### PR DESCRIPTION
**Pull request by CalvinWilkinson**
<h1 style="font-weight:bold" align="center">Preview Feature Pull Request</h1>
<h2 style="font-weight:bold" align="center">📋Review Checklist📋</h2>

- [x] The **_head(source)_** branch for this pull request is a **_preview feature_** branch, with the correct naming syntax, in the following example:
  - A **_head(source)_** branch is the branch being merged into a target branch.
    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/feature/<issue-num>-<description>
      Example: preview/feature/123-my-preview-feature
      ```
    </details>

- [x] The **_head(source)_** branch for this pull request is created from a **_preview release_** branch, with the correct naming syntax, in the following example:
  <details closed><summary>Example</summary>

    ``` xml
    Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
    Example: preview/v1.2.3-preview.4
    ```
  </details>

- [x] The **_base(target)_** branch for this pull request is a **_preview release_** branch, with the correct naming syntax, in the following example:
  - A **_base(target)_** branch is the branch that the **_head(source)_** branch is merging into.

    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
      Example: preview/v1.2.3-preview.4
      ```
    </details>

💡For more information on branching, refer to the project's [branching documentation](https://github.com/KinsonDigital/.github/blob/master/docs/BranchingFullGitFlow.md).

- [x] Associated issue exists and is linked to this pull request.
  - One issue per pull request.

- [x] Pull request title matches the title of the associated issue.

- [x] A `preview` label has been added to the pull request.

- [x] The labels attached to this PR match the labels attached to the associated issue.

- [x] Project assigned matches the project of the associated issue.

- [x] Milestone assigned matches the milestone of the associated issue.

- [x] My code follows the coding style of this project.
  - The style is enforced by the **_.editorconfig_** files in the project and displayed as warnings.  If there is an edge case, reach out, and let's discuss it.

- [x] I have manually tested my code changes to the best of my ability.

- [x] All tests passed locally.
  - This is required because unit tests might pass locally but not in the CI environment during the status check process or vice-versa.
  - Tests might pass on the developer's machine but not necessarily on the code reviewer's machine.
  - Status checks run for this pull request when merging into the **_preview/v\*.\*.\*-preview.\*_** branch.  These status checks run every time a change is pushed to the pull request.  These checks validate version syntax, tagging, builds, unit tests, and more.

### CODE REVIEW

**Comment by CalvinWilkinson** [comment]
> **Warning** Implemented https://github.com/KinsonDigital/Velaptor/labels/%F0%9F%A7%A8breaking%20changes
> - Refactored the `IRendererFactory.CreateRectangleRenderer()` method to IRendererFactory.CreateShapeRenderer()
> - Refactored the `RendererFactory.CreateRectangleRenderer()` method to RendererFactory.CreateShapeRenderer()
> - Refactored the `IRectangleRenderer` interface to `IShapeRenderer`

**Comment by CalvinWilkinson** [comment]
> **Note** Added https://github.com/KinsonDigital/Velaptor/labels/%E2%9C%A8new%20feature.
> Added a new `record struct` named `CircleShape` to be used for rendering circles.
> Added a new method overload to the `IShapeRenderer` interface and `ShapeRenderer` class with the signature `void Render(CircleShape circle, int layer = 0)`

File: Velaptor/Graphics/CircleShape.cs [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    /// </remarks>
+    public float Diameter
+    {
+        get => this.diameter;
+        set
+        {
+            value = value < 1f ? 1f : value;
+            this.diameter = value;
+        }
+    }
```
Comment (KinsonDigitalAdmin) [inline Velaptor/Graphics/CircleShape.cs:52]: Refactor code docs

```suggestion
    /// <summary>
    /// Gets or sets the diameter of the circle.
    /// </summary>
    /// <remarks>
    ///     The diameter is restricted to a minimum value of one.
    ///     <br/>
    ///     Changing the diameter will automatically change the <see cref="Radius"/>.
    /// </remarks>
    public float Diameter
    {
        get => this.diameter;
        set
        {
            value = value < 1f ? 1f : value;
            this.diameter = value;
        }
    }
```

File: Velaptor/Graphics/CircleShape.cs [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+    /// <summary>
+    /// Gets or sets the radius of the circle.
+    /// </summary>
+    /// <remarks>
+    /// This is half of the <see cref="Diameter"/>.
+    /// </remarks>
```
Comment (KinsonDigitalAdmin) [inline Velaptor/Graphics/CircleShape.cs:59]: Refactor docs

```suggestion
    /// <summary>
    /// Gets or sets the radius of the circle.
    /// </summary>
    /// <remarks>
    /// This is half of the <see cref="Diameter"/>.
    /// <br/>
    /// Changing the radius will automatically change the <see cref="Diameter"/>.
    /// </remarks>
```

File: Velaptor/Graphics/CircleShape.cs [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+    public float Radius
+    {
+        get => Diameter / 2f;
+        set => Diameter = value * 2f;
+    }
+
+    /// <summary>
+    /// Gets or sets the top location of the circle.
```
Comment (KinsonDigitalAdmin) [inline Velaptor/Graphics/CircleShape.cs:67]: Refactor code docs

```suggestion
    /// Gets or sets the location of the top of the circle on the Y axis.
```

File: Velaptor/Graphics/CircleShape.cs [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+    ///     Will automatically update the <see cref="Position"/> of the circle.
+    /// </remarks>
+    public float Top
+    {
+        get => Position.Y - Radius;
+        set => Position = new Vector2(Position.X, value + Radius);
+    }
+
+    /// <summary>
+    /// Gets or sets the right location of the circle.
```
Comment (KinsonDigitalAdmin) [inline Velaptor/Graphics/CircleShape.cs:79]: Refactor code docs

```suggestion
    /// Gets or sets the location of the right side of the circle on the X axis.
```

File: Velaptor/Graphics/CircleShape.cs [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+    ///     Will automatically update the <see cref="Position"/> of the circle.
+    /// </remarks>
+    public float Right
+    {
+        get => Position.X + Radius;
+        set => Position = new Vector2(value - Radius, Position.Y);
+    }
+
+    /// <summary>
+    /// Gets or sets the bottom location of the circle.
```
Comment (KinsonDigitalAdmin) [inline Velaptor/Graphics/CircleShape.cs:91]: Refactor code docs

```suggestion
    /// Gets or sets the location of the bottom of the circle on the Y axis.
```

File: Velaptor/Graphics/CircleShape.cs [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+    ///     Will automatically update the <see cref="Position"/> of the circle.
+    /// </remarks>
+    public float Bottom
+    {
+        get => Position.Y + Radius;
+        set => Position = new Vector2(Position.X, value - Radius);
+    }
+
+    /// <summary>
+    /// Gets or sets the Left location of the circle.
```
Comment (KinsonDigitalAdmin) [inline Velaptor/Graphics/CircleShape.cs:103]: Refactor code docs

```suggestion
    /// Gets or sets the location of the left side of the circle on the X axis.
```

File: Velaptor/Graphics/CircleShape.cs [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+    /// </summary>
+    /// <remarks>
+    ///     Ignored if the <see cref="GradientType"/> is set to any value other than <see cref="ColorGradient.None"/>.
+    /// </remarks>
+    public Color Color { get; set; } = Color.White;
+
+    /// <summary>
+    /// Gets or sets a value indicating whether or not the circle is filled or empty.
+    /// </summary>
+    public bool IsFilled { get; set; } = true;
```
Comment (KinsonDigitalAdmin) [inline Velaptor/Graphics/CircleShape.cs:125]: Refactor name of this property to `IsSolid`.

This is to prevent confusion with the `IsEmpty()` method which is for a different context.

```suggestion
    public bool IsSolid { get; set; } = true;
```

File: Velaptor/Graphics/RectShape.cs
Comment (KinsonDigitalAdmin) [inline Velaptor/Graphics/RectShape.cs:1]: Refactor the `IsFilled` property to the name of `IsSolid`

**Review by KinsonDigitalAdmin — CHANGES_REQUESTED** [review]

**Review by KinsonDigitalAdmin — APPROVED** [review]

**Comment by CalvinWilkinson** [comment]
> **Note** Fixed an issue with the `RectShape` border thickness not being maintained after increasing the width and height after the rectangle has been reduced to the minimum width and height dimensions.
